src/store/ducks/plans.ts

```diff
@@ -515,3 +518,150 @@ export function getPlanRecordsIdArray(
+ * @param props - the plan filters object
+ */
+export const getReason = (_: Registry, props: PlanFilters) => props.reason;
+
+/** getPlansArrayByInterventionType
+ * Gets an array of Plan objects filtered by interventionType
+ * @param state - the redux store
+ * @param props - the plan filters object
+ */
+export const getPlansArrayByInterventionType = createSelector(
+  [plansArrayBaseSelector, getInterventionType],
+  (plans, interventionType) =>
+    interventionType
+      ? plans.filter(plan => plan.plan_intervention_type === interventionType)
+      : plans
+);
+
+/** getPlansArrayByJurisdictionIds
+ * Gets an array of Plan objects filtered by jurisdictionIds
+ * @param state - the redux store
+ * @param props - the plan filters object
+ */
+export const getPlansArrayByJurisdictionIds = createSelector(
+  [plansArrayBaseSelector, getJurisdictionIds],
+  (plans, jurisdictionIds) =>
+    jurisdictionIds
+      ? plans.filter(plan =>
+          jurisdictionIds.length ? jurisdictionIds.includes(plan.jurisdiction_id) : true
+        )
+      : plans
+);
+
+/** getPlansArrayByStatus
+ * Gets an array of Plan objects filtered by plan status
+ * @param state - the redux store
+ * @param props - the plan filters object
+ */
+export const getPlansArrayByStatus = createSelector(
+  [plansArrayBaseSelector, getStatusList],
+  (plans, statusList) =>
+    statusList
+      ? plans.filter(plan => (statusList.length ? statusList.includes(plan.plan_status) : true))
+      : plans
+);
+
+/** getPlansArrayByReason
+ * Gets an array of Plan objects filtered by FI plan reason
+ * @param state - the redux store
+ * @param props - the plan filters object
+ */
+export const getPlansArrayByReason = createSelector(
+  [plansArrayBaseSelector, getReason],
+  (plans, reason) => (reason ? plans.filter(plan => plan.plan_fi_reason === reason) : plans)
+);
+
+/** getPlansArrayByParentJurisdictionId
+ * Gets an array of Plan objects filtered by plan jurisdiction parent_id
+ * @param state - the redux store
+ * @param props - the plan filters object
+ */
+export const getPlansArrayByParentJurisdictionId = createSelector(
+  [plansArrayBaseSelector, getParentJurisdictionId],
+  (plans, parentJurisdictionId) =>
+    plans.filter(
+      plan =>
+        (parentJurisdictionId && !plan.jurisdiction_path ? false : true) &&
+        (parentJurisdictionId && plan.jurisdiction_path
+          ? plan.jurisdiction_path.includes(parentJurisdictionId)
+          : true)
+    )
+);
+
+/** plansArraySelector
+ * Gets an array of Plan objects filtered by one or all of the following:
+ *    - interventionType
+ *    - jurisdictionIds
+ *    - plan status
+ *    - FI plan reason
+ *    - plan jurisdiction parent_id
+ *
+ * These filter params are all optional and are supplied via the prop parameter.
+ *
+ * This selector is meant to be a memoized replacement for getPlanRecordsArray.
+ *
+ * @param state - the redux store
+ * @param props - the plan filters object
```

Comment:
@moshthepitt Let's be sure to add the Typing for `props` here, especially for these created Selectors since the props/returns of the resultant Selector are not intuitive from what's being passed into `createSelector()`

Reply:
@cKellyDesign good catch!